### PR TITLE
Testing and expanded Arabic support

### PR DESCRIPTION
**Pull request by sgrieve**
This is a horrible PR which contains all of our work to get a stable version of the Arabic editing working, and the construction of a series of tests for the GUI.

@raquel-ucl I suggest we merge this together and build a new release from it on Thursday or Friday

🚀 🚀 🚀 🚀 🚀 🚀 🚀

### CODE REVIEW

**Comment by sgrieve** [comment]
The merge was simple, so I have done it. Once we have reviewed we can build and release the new versions.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Nammu: Desktop GUI for ORACC
 
 To run Nammu on your computer, you can download it from
-[here](https://github.com/oracc/nammu/releases/download/1.0.0/nammu-1.0.0.jar)
+[here](https://github.com/oracc/nammu/releases/download/1.2.0/nammu-1.2.0.jar)
```
[the remark this answers is not in the document]
Comment (sgrieve) [reply README.md]: Yes, thats what happened.

**Comment by raquel-ucl** [comment]
I've tested it for a bit on my machine now. I can build and open Nammu with no problems, and load Arabic files as well as display Arabic mode. And the "1" in the line number panel shows now! :) 
However undoing does seem to work, not even in non Arabic mode. I'm not sure about making a release if undoing is broken, because it freezes the edit pane. Let's discuss tomorrow, release might need to be delayed a few more days.

**Comment by raquel-ucl** [comment]
I've just run the tests locally and they all pass. I do occasionally at different times and with different colours this error:

```
python/nammu/test/test_nammu.py .........................Exception in thread "AWT-EventQueue-0" Traceback (most recent call last):
  File "/Users/raquelalegre/.virtualenvs/ORACC_Jython/Lib/site-packages/swingutils/threads/util.py", line 23, in run
    self._func(*self._args, **self._kwargs)
  File "/Users/raquelalegre/workspace/ORACC/nammu/python/nammu/controller/NammuController.py", line 223, in initHighlighting
    self.atfAreaController.syntax_highlight(top, bottom)
  File "/Users/raquelalegre/workspace/ORACC/nammu/python/nammu/controller/AtfAreaController.py", line 320, in syntax_highlight
    self.syntax_highlighter.syntax_highlight(top_line,
  File "/Users/raquelalegre/workspace/ORACC/nammu/python/nammu/view/SyntaxHighlighter.py", line 193, in syntax_highlight
    attribs = self.attribs[defaultcolor]
KeyError: 'black'
```

I'm OK with this breaking occasionally since it's re-evaluated all the time and won't be a lasting error, but it'd be nice to capture in a try/except block and write a log message instead.

File: python/nammu/controller/NammuController.py [hunk 2]
```diff
@@ -224,7 +241,12 @@ def saveFile(self, event=None):
         to save in desired location.
         Also checks for project name, and if found, makes it default.
         '''
-        atfText = self.atfAreaController.getAtfAreaText()
+
+        if self.arabic_edition_on:
+            atfText = self.atfAreaController.concatenate_arabic_text()
+        else:
+            atfText = self.atfAreaController.getAtfAreaText()
+        print(atfText)
```
[the remark this answers is not in the document]
Comment (sgrieve) [reply python/nammu/controller/NammuController.py]: Removed the rogue print statement and created the suggested method.

File: python/nammu/controller/NammuController.py [hunk 3]
```diff
@@ -785,11 +812,21 @@ def parse(self, text, event=None):
         else:
             return parsed
 
-    def unicode(self, event=None):
```
[the remark this answers is not in the document]
Comment (sgrieve) [reply python/nammu/controller/NammuController.py]: Reinstated this method.

File: python/nammu/controller/NammuController.py [hunk 4]
```diff
@@ -181,7 +198,7 @@ def openFile(self, event=None):
                 # Re-enable caret updating and syntax highlighting after load
                 self.atfAreaController.caret.setUpdatePolicy(
                                                     DefaultCaret.ALWAYS_UPDATE)
-                syntax_highlight.syntax_highlight_on = True
+                # syntax_high.syntax_highlight_on = True
```
[the remark this answers is not in the document]
Comment (sgrieve) [reply python/nammu/controller/NammuController.py]: Deleted.

File: .travis.yml [hunk 5]
```diff
@@ -1,25 +1,56 @@
...
-
-language: java
-
 matrix:
   include:
+    - python: 2.7
+      env: PYTEST=1
     - jdk: oraclejdk8
+      env: MVN=1
+    # - jdk: oraclejdk9
```
[the remark this answers is not in the document]
Comment (sgrieve) [reply .travis.yml]: We get the following error:

```
error: [Errno -1] Unmapped exception: org.python.netty.util.IllegalReferenceCountException: refCnt: 0
```
Which is also discussed in #302

The full travis log is here: https://travis-ci.org/oracc/nammu/jobs/413091852

File: python/nammu/controller/NammuController.py [hunk 6]
```diff
@@ -785,11 +812,21 @@ def parse(self, text, event=None):
...
-    def unicode(self, event=None):
+    def arabic(self, event=None):
         '''
-        Create bool for unicode, change value when clicked.
+        Create bool for arabic, change value when clicked.
         '''
-        self.logger.debug("Unicode...")
+        self.logger.debug("Enabling/Disabling arabic translation mode...")
+        if event:
+            # todo: handle unsaved arabic area
```
[the remark this answers is not in the document]
Comment (sgrieve) [reply python/nammu/controller/NammuController.py]: Testing this it seems like the translation pane is cleared, without a save prompt but we do not revert back to the single pane view.

Working to resolve this now.

**Comment by sgrieve** [comment]
We now capture that KeyError and write a debug message.

File: README.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # Nammu: Desktop GUI for ORACC
```
[the remark this answers is not in the document]
Comment (sgrieve) [reply README.md]: The readme has been updated to cover the new arabic translation pane, and a new screenshot has been added.

**Pull request merged**